test/polydb.jl

```diff
@@ -0,0 +1,163 @@
+using Mongoc
+
+@testset "polyDB" begin
+    add_constraints_poly = get(ENV, "POLYDB_SERVER_URI", "") == "" ? ["DIM" => Dict("\$lte" => 3)] : []
+    function _acp(a::Array)
+        return append!(Array{Pair{String,Any}}(a), add_constraints_poly)
+    end
+    add_constraints_mat = get(ENV, "POLYDB_SERVER_URI", "") == "" ? [:("N_ELEMENTS" <= 4)] : []
+    function _acm(a::Array)
+        return append!(Array{Expr}(a), add_constraints_mat)
+    end
+
+    @testset "Basic functionality" begin
+        @test Polymake.Polydb.get_db() isa Polymake.Polydb.Database
+        db = Polymake.Polydb.get_db()
+        @test db["Polytopes.Lattice.SmoothReflexive"] isa Polymake.Polydb.Collection
+        @test db["Polytopes.Lattice.SmoothReflexive"] isa Polymake.Polydb.Collection{Polymake.BigObject}
+        try
+            @test Mongoc.ping(db.mdb.client)["ok"] == 1
+        catch
+            @test "not" == "connected"
+        end
+        collection_bo = db["Polytopes.Lattice.SmoothReflexive"]
+        @test Polymake.Polydb.Collection{Mongoc.BSON}(collection_bo) isa Polymake.Polydb.Collection
+        @test Polymake.Polydb.Collection{Mongoc.BSON}(collection_bo) isa Polymake.Polydb.Collection{Mongoc.BSON}
+        collection_bson = Polymake.Polydb.Collection{Mongoc.BSON}(collection_bo)
+        @test Polymake.Polydb.Collection{Mongoc.BSON}(collection_bo) isa Polymake.Polydb.Collection
+        @test Polymake.Polydb.Collection{Mongoc.BSON}(collection_bo) isa Polymake.Polydb.Collection{Mongoc.BSON}
```

Comment:
we're testing the same thing twice here, is it intended?

I guess the first two tests should be with `Polymake.BigObject`, then the second two tests with `Mongoc.BSON` and then define `collection_bson`?